main.cpp

```diff
@@ -0,0 +1,149 @@
+#include "ProductCatalog.h"
+#include "InventoryManager.h"
+#include "CustomExceptions.h"
+#include <iostream>
+#include <limits> // for clearing input buffer
+using namespace std;
+
+void clearInputBuffer() {
+    cin.clear();
+    cin.ignore(numeric_limits<streamsize>::max(), '\n');
+}
+
+int main() {
+    ProductCatalog catalog;
+    InventoryManager manager;
+
+    int choice;
+    bool exitProgram = false;
+
+    do {
+        try {
+            cout << "Main Menu:\n";
+            cout << "1. Add Product\n";
+            cout << "2. Update Product\n";
+            cout << "3. Delete Product\n";
+             cout << "4. Sell Product\n";
+            cout << "5. Display Inventory\n";
+            cout << "6. Exit\n";
+            cout << "Enter your choice: ";
+            cin >> choice;
+
+            switch (choice) {
+                case 1: {
+                    int id;

```

Comment:
Can create separate methods for each case

Reply:
Done